### PR TITLE
apply clang tidy fixes

### PR DESCRIPTION
**Pull request by tjhei**

### CODE REVIEW

File: source/postprocess/depth_average.cc [hunk 1]
```diff
@@ -164,8 +164,8 @@ namespace aspect
             {
               if (output_format_string != "txt")
                 {
-                  for (unsigned int j=0; j<variables.size(); ++j)
-                    data_out_stack.declare_data_vector (variables[j],
+                  for (auto &variable : variables)
```
Comment (gassmoeller) [inline source/postprocess/depth_average.cc:167]:
```suggestion
                  for (const auto &variable : variables)
```

File: source/postprocess/depth_average.cc [hunk 2]
```diff
@@ -233,8 +233,8 @@ namespace aspect
 
                   // Write the header
                   f << "#       time" << "        depth";
-                  for ( unsigned int i = 0; i < variables.size(); ++i)
-                    f << ' ' << variables[i];
+                  for (auto &variable : variables)
```
Comment (gassmoeller) [inline source/postprocess/depth_average.cc:236]:
```suggestion
                  for (const auto &variable : variables)
```

File: source/postprocess/geoid.cc [hunk 3]
```diff
@@ -809,7 +809,7 @@ namespace aspect
           std::vector<double> gravity_anomaly;
           gravity_anomaly.reserve(surface_cell_spherical_coordinates.size());
 
-          for (unsigned int i=0; i<surface_cell_spherical_coordinates.size(); ++i)
+          for (auto &surface_cell_spherical_coordinate : surface_cell_spherical_coordinates)
```
Comment (gassmoeller) [inline source/postprocess/geoid.cc:812]:
```suggestion
          for (const auto &surface_cell_spherical_coordinate : surface_cell_spherical_coordinates)
```

File: source/postprocess/global_statistics.cc [hunk 4]
```diff
@@ -131,8 +131,8 @@ namespace aspect
       // vectors in which we store our data.
       unsigned int nonlinear_iterations = stokes_iterations_cheap.size();
 
-      for (unsigned int column=0; column<advection_iterations.size(); ++column)
-        nonlinear_iterations = std::max(nonlinear_iterations, static_cast<unsigned int> (advection_iterations[column].second.size()));
+      for (auto &advection_iteration : advection_iterations)
```
Comment (gassmoeller) [inline source/postprocess/global_statistics.cc:134]:
```suggestion
      for (const auto &advection_iteration : advection_iterations)
```

File: source/postprocess/global_statistics.cc [hunk 5]
```diff
@@ -142,10 +142,10 @@ namespace aspect
             statistics.add_value("Nonlinear iteration number",
                                  iteration);
 
-            for (unsigned int column=0; column<advection_iterations.size(); ++column)
-              if (iteration < advection_iterations[column].second.size())
-                statistics.add_value(advection_iterations[column].first,
-                                     advection_iterations[column].second[iteration]);
+            for (auto &advection_iteration : advection_iterations)
```
Comment (gassmoeller) [inline source/postprocess/global_statistics.cc:145]:
```suggestion
            for (const auto &advection_iteration : advection_iterations)
```

File: source/postprocess/interface.cc [hunk 6]
```diff
@@ -281,8 +281,8 @@ namespace aspect
                                        ">, but the latter is not a valid name."));
 
               bool already_present = false;
-              for (unsigned int n=0; n<postprocessor_names.size(); ++n)
-                if (postprocessor_names[n] == p)
+              for (auto &postprocessor_name : postprocessor_names)
```
Comment (gassmoeller) [inline source/postprocess/interface.cc:284]:
```suggestion
              for (const auto &postprocessor_name : postprocessor_names)
```

File: source/postprocess/visualization.cc [hunk 7]
```diff
@@ -1382,11 +1382,11 @@ namespace aspect
 
       // then go through the list, create objects and let them parse
       // their own parameters
-      for (unsigned int name=0; name<viz_names.size(); ++name)
+      for (auto &viz_name : viz_names)
```
Comment (gassmoeller) [inline source/postprocess/visualization.cc:1385]:
```suggestion
      for (const auto &viz_name : viz_names)
```

File: source/postprocess/visualization/named_additional_outputs.cc [hunk 8]
```diff
@@ -64,8 +64,8 @@ namespace aspect
               {
                 std::vector<std::string> names = result->get_names();
 
-                for (unsigned int i=0; i<names.size(); ++i)
-                  property_names.push_back(names[i]);
+                for (auto &name : names)
```
Comment (gassmoeller) [inline source/postprocess/visualization/named_additional_outputs.cc:67]:
```suggestion
                for (const auto &name : names)
```

File: source/termination_criteria/interface.cc [hunk 9]
```diff
@@ -227,18 +227,18 @@ namespace aspect
 
       // go through the list, create objects, initialize them, and let them parse
       // their own parameters
-      for (unsigned int name=0; name<plugin_names.size(); ++name)
+      for (auto &plugin_name : plugin_names)
```
Comment (gassmoeller) [inline source/termination_criteria/interface.cc:230]:
```suggestion
      for (const auto &plugin_name : plugin_names)
```

File: source/time_stepping/interface.cc [hunk 10]
```diff
@@ -342,11 +342,11 @@ namespace aspect
 
         prm.leave_subsection();
 
-        for (unsigned int name=0; name<model_names.size(); ++name)
+        for (auto &model_name : model_names)
```
Comment (gassmoeller) [inline source/time_stepping/interface.cc:345]:
```suggestion
        for (const auto &model_name : model_names)
```

File: source/volume_of_fluid/reconstruct.cc [hunk 11]
```diff
@@ -321,9 +321,9 @@ namespace aspect
               const std::vector<double> weights = fevalues.get_JxW_values();
 
               double cell_vol = 0.0;
-              for (unsigned int j=0; j<weights.size(); ++j)
+              for (double weight : weights)
```
Comment (gassmoeller) [inline source/volume_of_fluid/reconstruct.cc:324]:
```suggestion
              for (const double weight : weights)
```

File: source/volume_of_fluid/reconstruct.cc [hunk 12]
```diff
@@ -359,9 +359,9 @@ namespace aspect
                 const std::vector<double> weights = fevalues.get_JxW_values();
 
                 double cell_vol = 0.0;
-                for (unsigned int j=0; j<weights.size(); ++j)
+                for (double weight : weights)
```
Comment (gassmoeller) [inline source/volume_of_fluid/reconstruct.cc:362]:
```suggestion
                for (const double weight : weights)
```

**Review by gassmoeller — COMMENTED** [review]
Looks good, thanks for the cleanup. I think there are a bunch of places where variables can be `const`. I think I marked all of them with suggestions. Can you take a look?

**Review by MFraters — APPROVED** [review]
Looks good to me. I like the modernized for loops, they look a lot cleaner to me. I don't really understand why it prefers static_cast<..> and reinterpret_cast<...> over just (..), but I am fine with either.

**Comment by tjhei** [comment]
> I don't really understand why it prefers static_cast<..> and reinterpret_cast<...> over just (..), but I am fine with either.

``(bla*)`` is a C cast and people prefer one of the c++ casts instead. ``Bytef`` from zlib is an ``unsigned char``, while we work with ``char``, so we do need a ``reinterpret_cast`` and a ``static_cast`` won't work.

**Pull request merged**